src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -480,22 +494,35 @@ private static boolean loadFile(Model dataToBeValidated, File fileToValidate) {
     }
 
     // Dump the validation report model either to stderr or a file, and return the number of violations.
-    private static int processReport(Resource validationReport, File fileToValidate, boolean fileOutput) {
+    private static int processReports(Set<Resource> validationReports, File fileToValidate, boolean fileOutput) {
         if (!fileOutput) {
-            logger.info("---> Validation report:");
-            RDFDataMgr.write(System.err, validationReport.getModel(), RDFFormat.TURTLE_PRETTY);
+            logger.info("---> Validation report(s):");
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            for (Resource report : validationReports) {
+                RDFDataMgr.write(baos, report.getModel(), RDFFormat.TURTLE_PRETTY);
+            }
+            logger.info(baos.toString());
         } else {
-            String outputFilename = fileToValidate.toString().replace(".ttl", "-report.txt");
-            try {
-                RDFDataMgr.write(Files.newOutputStream(Paths.get(outputFilename)),
-                        validationReport.getModel(), RDFFormat.TURTLE_PRETTY);
-            } catch (IOException ioe) {
-                logger.warn("---> Could not write validation report for " + fileToValidate + ".");
+            String suffix = validationReports.size() == 1 ? "-report.txt" : "-report-%d.txt";
+            String template = fileToValidate.toString().replace(".ttl", suffix);
+            int i = 1;
+            for (Resource report : validationReports) {
+                String outputFilename = String.format(template, i++);
```

Comment:
Each report is from a particular thread, which is why they're numbered rather than providing information about their content. We don't have control over which shapes/constraints go to particular threads, so we'd have to inspect each model before creating the file. Let me know what kind of naming convention you were thinking about

Reply:
I can't think of a better naming convention.  But this multiple report setup should probably be documented in the README somewhere as it's probably unexpected behavior.

Reply:
I'm fine with leaving it as is and documenting it.

Reply:
It's also possible to force the validator to create only a single file, but if the file has many validations, this could take a while to stitch the thread models together. Let me know what you guys would like to do.

Reply:
I'd prefer to get it in as is, and then defer stitching the thread models back together to a future enhancement.

Reply:
To clarify, by "stitch the thread models together" are you talking about simply having the code call `validateKBAndReturnReport()` instead of `validateKBAndReturnMultipleReports()`, and the "it could take a while" means validation time, not development time?  I think Craig and I were under the impression meant you could stitch the models together in code, creating a single report, but that will take some development time and delay this PR from getting merged into master-- hence why he suggested it as a future enhancement.

Reply:
Yes, validateKBAndReturnReport() already does what you want. And yes, calling it instead of validateKBAndReturnMultipleReports() will take more validation time, not development time.

Reply:
Any idea how much longer validation takes?  I don't know if you have any metrics on it.  I did a quick test of a shallow validation (t=1000) of an invalid scaling test with 10,000 entities and the validation time was just about the same.  I'm sure that for larger/more complicated performer KBs, that it will absolutely take longer to combine the models in code.

Reply:
Related to the above, maybe it could use `validateKBAndReturnReport()` when using shallow validation or fail-fast, when numbers of violations should be more moderate.

Reply:
Given the confusion this is causing, maybe it's better to add a flag to make multiple reports optional

Reply:
Although I wouldn't object to a flag to force single-file reports (I would find it convenient) I don't know if it's something a performer would care about, nor do we need it for TA2 validation.  Maybe the single/multiple report format could be mentioned in the README in the `Running the validator in code` section, and we can defer anything else to later if necessary.

Reply:
I'm fine with leaving it as is, mentioning it in the README, and deferring anything else to later.

Reply:
README updated.